cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/TetheringClientHandler.java

```diff
@@ -81,4 +84,15 @@ private void validateNamespaces(List<NamespaceAllocation> namespaceAllocations)
       }
     }
   }
+
+  private void validateEndpoint(String endpoint) throws BadRequestException {
+    if (endpoint == null) {
+      throw new BadRequestException("Endpoint must be specified");
+    }
+    try {
+      new URL(endpoint).toURI();
+    } catch (Exception e) {
+      throw new BadRequestException(String.format("Endpoint %s is not a valid URI", endpoint));
```

Comment:
would be good to include the exception message to give some idea of what is wrong `URI: %s", e.getMessage()`. 


Also include e as the cause

Reply:
fixed